sdk/core/azure-core/src/main/java/com/azure/core/util/ClientOptions.java

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.util;
+
+import com.azure.core.annotation.Fluent;
+import com.azure.core.util.logging.ClientLogger;
+
+import java.util.ArrayList;
+import java.util.Objects;
+
+/**
+ * Client Options for setting common properties for example applicationId. Most of these properties are applied on
+ * request being send to Azure Service but some could be used for other purpose also for example applicationId could be
+ * used for telemetry.
+ */
+@Fluent
+public final class ClientOptions {
```

Comment:
Mind giving an example on how this class would be used? Is it expected to be passed during the construction of an SDK client or for each service call made?

Right now, `UserAgentPolicy` has an immutable `User-Agent` string, so it feels like this would be passed during client construction only?

Reply:
This would be set  in Builder and client creation time.